packages/adapter-components/src/elements/instance_elements.ts

```diff
@@ -103,10 +106,22 @@ export const toBasicInstance = async ({
     ? fileNameParts.join('_')
     : undefined))
 
-  const naclName = naclCase(
+  const desiredName = naclCase(
     parent && nestName ? `${parent.elemID.name}${ID_SEPARATOR}${name}` : String(name)
   )
   const adapterName = type.elemID.adapter
+  const naclName = getElemIdFunc && serviceIdField
+    ? getElemIdFunc(
+      adapterName,
+      {
+        [serviceIdField]: entry[serviceIdField],
+        [OBJECT_SERVICE_ID]: toServiceIdsString({
+          [OBJECT_NAME]: type.elemID.getFullName(),
+        }),
+      },
+      desiredName
+    ).name
+    : desiredName
```

Comment:
style nit - there are too many lines here for a `? :` operator
maybe split out the "createServiceIds" part to a function?
```suggestion
  const naclName = getElemIdFunc && serviceIdField
    ? getElemIdFunc(adapterName, createServiceIds(entry, serviceIdField), desiredName).name
    : desiredName
```